src/main/java/org/variantsync/diffdetective/variation/diff/bad/BadVDiff.java

```diff
@@ -0,0 +1,392 @@
+        ) {}
+
+        final FromGoodNodeTranslation nodeTranslation = new FromGoodNodeTranslation();
+
+        final Map<VariationTreeNode, VariationTreeNode>   matching = new HashMap<>();
+        final Map<VariationTreeNode, DiffType>            coloring = new HashMap<>();
+        final Map<VariationTreeNode, DiffLineNumberRange> lines    = new HashMap<>();
+        final Set<DiffNode> splittedNodes = new HashSet<>();
+
+        final List<EdgeToConstruct> edgesToConstruct = new ArrayList<>();
+
+        final VariationTreeNode root = fromGood(d.getRoot(), nodeTranslation, coloring, lines);
+
+        d.forAll(diffNode -> {
+            if (diffNode == d.getRoot()) {
+                return;
+            }
+
+            final DiffNode pbefore = diffNode.getParent(BEFORE);
+            final DiffNode pafter = diffNode.getParent(AFTER);
+
+            final DiffType color = diffNode.getDiffType();
+
+            final boolean unchanged = color == NON;
+            final boolean hasUnsplitUnchangedParent =
+                       pbefore == pafter
+                    && pbefore != null
+                    && pbefore.isNon()
+                    && !splittedNodes.contains(pbefore);
+            /*
+             * We split every node that is unchanged and whose parent was also splitted.
+             * In fact, we only have to split unchanged nodes but the second clause makes the bad diff
+             * a bit less bad by splitting unchanged nodes only when necessary.
+             * Basically, a variation tree can represent unchanged nodes when there were no changes
+             * above the unchanged node.
+             * By removing the second clause, we obtain a bad diff that splits in exactly the two
+             * projections of the variation diff.
+             */
+            final boolean split = unchanged && !hasUnsplitUnchangedParent;
+            if (split) {
+                Assert.assertTrue(pbefore != null && pafter != null);
+
+                final VariationTreeNode selfBefore = plain(diffNode);
+                final VariationTreeNode selfAfter  = plain(diffNode);
+
+                nodeTranslation.put(diffNode, BEFORE, selfBefore);
+                nodeTranslation.put(diffNode, AFTER,  selfAfter);
+
+                edgesToConstruct.add(new EdgeToConstruct(
+                        selfBefore, pbefore, BEFORE
+                ));
+                edgesToConstruct.add(new EdgeToConstruct(
+                        selfAfter, pafter, AFTER
+                ));
+
+                // further metadata to copy
+                final DiffLineNumberRange dRange = new DiffLineNumberRange(diffNode.getFromLine(), diffNode.getToLine());
+                lines.put(selfBefore, dRange);
+                lines.put(selfAfter,  dRange);
+                coloring.put(selfBefore, REM);
+                coloring.put(selfAfter,  ADD);
+                matching.put(selfBefore, selfAfter);
+                matching.put(selfAfter,  selfBefore);
```

Comment:
You can get rid of half of this code by using `Time.forAll`.

Reply:
I guess you are right. This would however require to out selfBefore and selfAfter (as well as pbefore and pafter) into maps to index them by time. I am not sure if that would simplify things here.

Reply:
You're right I missed that (although `pbefore` and `pafter` can be replaced by `diffNode.getParent(time)`). For comparison I implemented this change in 8165c9bee0b3e97b085d6d3ce2954f88dece09c4. I'm also not sure if this is actually better but I prefer to factor out such duplicate code.
A method might be better in this case. What do you think.

Reply:
I like your patch. I will merge it into this PR. :)